compiler-lib/src/type_checking/mod.rs

```diff
@@ -1,409 +1,32 @@
-use crate::{
-    asciifile::{Span, Spanned},
-    ast,
-    context::Context,
-    semantics::SemanticError,
-    strtab::{StringTable, Symbol},
-};
-use std::{
-    collections::{HashMap, HashSet},
-    hash::Hash,
-};
-
+//! This module implements type checking on the AST.
+//! The type checking produces an instance of `type_system::TypeSystem`, which
+//! is later used by the `firm` module for IR graph construction.
+//!
+//!
+//! ## `$` Types (`builtin_types`)
+//!
+//! MiniJava suffers from the fact that its specification was stripped down
+//! from the Java language report.
+//! A good example is the handling of the main method's arguments:
+//! `String[] args` must not be used, and according to the spec, `String[]`
+//! within the main method's parameter list is not a type.
+//! However, it is possible for users to define a `class String` in MiniJava,
+//! which ,in Java, would shadow the `java.lang.String` that is part of the
+//! main method's parameter type. Not so in MiniJava - at least that's the
+//! result of the discussion we had in the
```

Comment:
NIT: @problame I think you have the same problem as me. Your default editor config brakes lines shorter than our rust format config, which results in these weird line breaks within paragraphs :)

Reply:
Thanks for the docs